### DISCUSSION TITLE
v1.0.0-beta4

### DISCUSSION THREAD

**Discussion by Mrs4s** [Mrs4s/go-cqhttp#898] [Show and tell]
新增: 企点协议的初步支持. [文档](https://github.com/Mrs4s/go-cqhttp/blob/master/docs/config.md#设备信息)
新增: `/qidian_get_account_info` API. 可获取企点账号信息. [文档](https://github.com/Mrs4s/go-cqhttp/blob/master/docs/cqhttp.md#获取企点账号信息)
新增: `/delete_friend` API. 用于主动删除好友. [文档](https://github.com/Mrs4s/go-cqhttp/blob/master/docs/cqhttp.md#删除好友)
新增: `/_get_model_show` API. 用于获取好友在线机型展示信息. by @MikeWang000000 
新增: `/_set_model_show` API. 用于设置好友在线机型展示信息. by @MikeWang000000 
新增: `/create_group_file_folder` API. 用于新建群文件文件夹.
新增: `/delete_group_folder` API. 用于删除群文件文件夹
新增: `/delete_group_file` API. 用于删除群文件.
新增: `-w` 启动参数. 用于覆盖工作目录. by @wdvxdr1123 
新增: 收到消息后会自动标记已读了.
新增: `/_send_group_notice` API支持通过 `image` 参数设置图片了. by @wdvxdr1123 
新增: 支持在不启动 `http监听` 的情况下启动 `http post` 上报了. by @wdvxdr1123 
新增: 为自动更新添加了校验支持. by @wdvxdr1123 
新增: 配置文件生成器. 在第一次启动时能帮助生成配置文件. by @wdvxdr1123 

修复: 扫码登录时可能会出现 `Packet timed out` 的问题. (有待进一步测试)
修复: 部分情况下私聊消息可能会重复接受的问题. (有待进一步测试)
修复: 检查更新会错误的将 `Backport` 版本当作最新版的问题. by @wdvxdr1123 
修复: 私聊信息的 `reply` 消息段的ID可能会出现错误的问题. by @wdvxdr1123 
修复: 在特殊情况下, 读取缓存文件不会释放的问题. by @wdvxdr1123 
修复: 在包含 `ExtraReplyData` 的情况下, `seq` 和 `time` 参数无法正常应用的问题. by @wdvxdr1123 
修复: `[CQ:tts]` 消息在 `iOS` 设备无法正常播放的问题. by @wdvxdr1123 
修复: 在会话过期时可能会陷入重连死循环的问题.
修复: 私聊情况下, 自身发送信息以`message`而非`message_sent`上报. by @sam01101  

优化: 优化了部分请求和功能时的内存占用. by @wdvxdr1123 
优化: 大幅度优化了解析 `base64` 时的内存使用. by @wdvxdr1123 
优化: 优化了多媒体文件的上传速度. by @wdvxdr1123 
优化: 所有的图片资源都更新到了 `https`. by @takayama-lily 
优化: 优化了内置数据库的磁盘空间和内存占用. by @wdvxdr1123 
优化: 优化了API错误时的返回文本. by @mnixry 

<hr /><em>This discussion was created from the release <a href='https://github.com/Mrs4s/go-cqhttp/releases/tag/v1.0.0-beta4'>v1.0.0-beta4</a>.</em>

**Comment by Yao990x16** [comment]
新增: 收到消息后会自动标记已读了.  这个有关掉的参数嘛?
